dataland-frontend/tests/e2e/fixtures/lksg/LksgDataFixtures.ts

```diff
@@ -0,0 +1,174 @@
+import { faker } from "@faker-js/faker";
+import { LksgData, ProductionSite } from "../../../../build/clients/backend";
+
+import { randomYesNoUndefined } from "../common/YesNoFixtures";
+
+export function generateProductionSite(): ProductionSite {
+  const fakeSiteName = faker.company.name();
+  const yesNo = randomYesNoUndefined();
+  const fullFormattedFakeAddress =
+    faker.address.street() +
+    " " +
+    faker.address.buildingNumber() +
+    ", " +
+    faker.address.zipCode() +
+    " " +
+    faker.address.city() +
+    ", " +
+    faker.address.country();
+  const fakeGoodsAndServices = Array.from(
+    { length: faker.datatype.number({ min: 1, max: 5 }) },
+    faker.commerce.productName
+  );
+
+  return {
+    name: fakeSiteName,
+    isInHouseProductionOrIsContractProcessing: yesNo,
+    address: fullFormattedFakeAddress,
+    listOfGoodsAndServices: fakeGoodsAndServices,
+  };
+}
+
+export function generateArrayOfProductionSites(): ProductionSite[] {
+  return Array.from({ length: faker.datatype.number({ min: 1, max: 50 }) }, generateProductionSite);
+}
+
+export function generateDataDate(): string {
+  return faker.date.future(1).toISOString().split("T")[0];
+}
+
+export function generateVatIdentificationNumber(): string {
+  const fakeCountryCode = faker.address.countryCode();
+  const randomNineDigitNumber = faker.random.numeric(9);
+  return fakeCountryCode + randomNineDigitNumber.toString();
+}
+
+export function getCompanyLegalForm(): string {
+  const legalForms = [
+    "Public Limited Company (PLC)",
+    "Private Limited Company (Ltd)",
+    "Limited Liability Partnership (LLP)",
+    "Partnership without Limited Liability",
+    "Sole Trader",
+    "GmbH",
+    "AG",
+    "GmbH & Co. KG",
+  ];
+  return legalForms[Math.floor(Math.random() * legalForms.length)];
+}
+
+export function generateIso4217CurrencyCode() {
+  const someCommonIso4217CurrencyCodes = ["USD", "EUR", "CHF", "CAD", "AUD"];
+  return someCommonIso4217CurrencyCodes[Math.floor(Math.random() * someCommonIso4217CurrencyCodes.length)];
+}
+
+export function generateLksgData(): LksgData {
+  const returnBase: LksgData = {};
+
+  returnBase.betterWorkProgramCertificate = randomYesNoUndefined();
+  returnBase.dataDate = generateDataDate();
+  returnBase.companyLegalForm = getCompanyLegalForm();
+  returnBase.vatIdentificationNumber = generateVatIdentificationNumber();
+  returnBase.numberOfEmployees = faker.datatype.number({ min: 1000, max: 200000 });
+  returnBase.shareOfTemporaryWorkers = faker.datatype.number({ min: 0, max: 30, precision: 0.01 });
+  returnBase.totalRevenue = faker.datatype.float({ min: 10000000, max: 100000000000 });
+  returnBase.totalRevenueCurrency = generateIso4217CurrencyCode();
+  returnBase.responsibilitiesForFairWorkingConditions = randomYesNoUndefined();
```

Comment:
I am not a fan of this huge block of code. How about putting all the randomYesNoUndefined values into a loop. Something like for each yesNoNa value in the backend model for LkSG create an entry with the same name and a yesNoNa value in the LksgData object?

Reply:
Emanuel and I had a look into this and it turns out it is not possible to do this using typescript